resolver_data_entry_server/db/query-controller/global-lock.js

```diff
@@ -0,0 +1,24 @@
+const sql = require('mssql');
+const { makeDBConnectionAndPS } = require('../dbUtils');
+const utils = require('../../bin/resolver_utils');
+
+// function for getting the global lock for write operation
+const getGlobalWritingLock = async () => {
+  try {
+    const ps = await makeDBConnectionAndPS();
+    await ps.prepare("select * from global_locks where name = 'write'");
+    const queryResponse = await ps.execute({});
+    utils.logThis('Successfully fetch global lock from DB');
+    await ps.unprepare();
+
+    return queryResponse.recordset;
+  } catch (error) {
+    utils.logThis('Error to connect DB for getting global_locks');
+  }
+};
+
+// DB query to r
+
+module.exports = {
+  getGlobalWritingLock,
+};
```

Comment:
I think we should use the environment variable approach instead of the database table approach. This is because setting an environment variable is a straightforward way to control application behavior, and it allows for decoupling your application logic from the database schema.

With your current approach, every time you write data to the database, it requires an extra query to check the `lock` flag before proceeding.

By using the environment variable approach, there is no need for a script to migrate the database for the `global_locks` table.

I think the better and easier way to prevent writing to the database is by using an environment variable.

Reply:
Using the environment variable will have a problem with the live product, every time we want to unlock/lock the data, we have to redeploy